extensions/smallrye-context-propagation/deployment/src/main/java/io/quarkus/smallrye/context/deployment/SmallRyeContextPropagationProcessor.java

```diff
@@ -86,4 +106,239 @@ void build(SmallRyeContextPropagationRecorder recorder,
                         .supplier(recorder.initializeManagedExecutor(executorBuildItem.getExecutorProxy()))
                         .setRuntimeInit().done());
     }
+
+    // transform IPs for ManagedExecutor/ThreadContext that use config annotation and don't yet have @NamedInstance
+    @BuildStep
+    InjectionPointTransformerBuildItem transformInjectionPoint() {
+        return new InjectionPointTransformerBuildItem(new InjectionPointsTransformer() {
+            @Override
+            public boolean appliesTo(Type requiredType) {
+                return true;
```

Comment:
Why don't we match the `ManagedExecutor/ThreadContext` here?

Reply:
Ah, that's a leftover from previous version. Without `BeanDiscoveryFinishedBuildItem` I actually needed this information inside IP processing but I can rework this now. Good call, thanks!